### ISSUE TITLE
Can not reproduce your precision in paper

### ISSUE THREAD

**Issue by whu-lyh** [PRBonn/kppr#2]
Hi, thanks for your great job first. Recently, I tried your codes with your official compressed data. Here is my setting:
```
aggregation:
  method: vlad
  vlad:
    feature_dim: 256
    nr_center: 64
    out_dim: 256
batch_size: 11
data_config:
  batch_size: __11__
  data_dir: ../data/oxford_compressed/
  dataset_loader: OxfordEmbeddingPad
  get_negatives: false
  num_negatives: 18
  num_positives: 2
  num_worker: __8__
  test_queries: oxford_compressed/test_queries_baseline.pickle
  train_queries: oxford_compressed/training_queries_baseline.pickle
experiment:
  id: tmp
feature_bank: 15000
git_commit_version: b'0708c1f'
kpconv:
  f_dscale: 2
  in_channels: 256
  kernel_size: 3
  num_layer: 7
  num_neighbors: 16
  out_channels: 256
  precompute_weights: true
  radius: 0.05
loss:
  params:
    alpha: 0.3
    margin: 0.5
  type: entropy
network_architecture: KPPR
point_net:
  in_dim: 6
  out_dim: 256
train:
  lr: 1.0e-05
  max_epoch: 40
  n_gpus: 1
```
I set the environment with your installation scripts, and the version is same as yours. I aligned the hyper-parameters with the recommendations, except the batch_size and the num_worker. Besides I set the padding point number to 4096, which is set to 2000 at [here](https://github.com/PRBonn/kppr/blob/edcd7f585e569cea34a3367efe8a33cba84a0a8e/kppr/datasets/datasets.py#L150). I train 20 epoch on 4090 with 24 GB memory, and get the following recalls:
![image](https://github.com/PRBonn/kppr/assets/32642500/9509f19a-2974-4494-a037-692df85c9261)
and 
![image](https://github.com/PRBonn/kppr/assets/32642500/2a136920-8535-4302-9186-fa135309aee1)
Is there anything wrong? Thanks again!

**Comment by louis-wiesmann** [comment]
Be aware that the numbers on tensorboard are not the final metrics. These are approximations but are different than under real experiment conditions. For the evaluation one is taking a single route and tries to localize within this with a different run from a different point in time.
The validation recall for tensorboard are computed based on the feature bank. And by this the point clouds are computed from all possible timestamps. Be aware that we keep an own [validtation feature bank](https://github.com/PRBonn/kppr/blob/main/kppr/models/models.py#L51) with fix size.
